File: Employee/src/main/java/EMS/Employee/services/EmployeeService.java
package EMS.Employee.services;

import java.util.List;

import EMS.Employee.model.Employee;

public interface EmployeeService {

	Employee createEmployee(Employee employee);

	List<Employee> getAllEmployees();

	boolean deleteEmployee(long id);

	Employee getEmployeeById(long id);

	Employee updateEmployee(long id, Employee employee);

}

File: employee-ui/src/services/EmployeeService.js
import axios from "axios";

const EMPLOYEE_BASE_URL = "http://localhost:8080/api/v1/employees"

class EmployeeService{
    saveEmployee(employee){
        return axios.post(EMPLOYEE_BASE_URL, employee);
    }

    getEmployees(){
        return axios.get(EMPLOYEE_BASE_URL);
    }

    deleteEmployee(id){
        return axios.delete(EMPLOYEE_BASE_URL + "/" + id);
    }

    getEmployeeById(id){
        return axios.get(EMPLOYEE_BASE_URL + "/" + id);
    }

    updateEmployee(employee, id){
        return axios.put(EMPLOYEE_BASE_URL + "/" + id, employee);
    }
}

export default new EmployeeService();